### PR TITLE
Setup settings

### PR DESCRIPTION
**Pull request by urwrstkn8mare**
I'm not sure if the `enforce-admins` property will actually stop us from making changes without reviews and also if the settings will apply to this repo. As far as I'm aware this only protects the default branch from merging PR without required reviews. I did not enable auto merge. I think admins still have power to remove the bot and stuff so I'm not sure how useful this will be aside from the obvious convenience of having the branch protections and some basic settings setup automatically when creating a new repo. This PR is only a draft as I'm not experienced with this side of things so please feel free to help out (see https://github.com/github/safe-settings) for what this is for.

### CODE REVIEW

File: settings.yml
Comment (UncleGoogle) [inline settings.yml:60]: It can be useful along with `allow_merge_commit: false` for contributors that does not know git well. I would keep true for now.

**Comment by AndrewDWhite** [comment]
A quick skim of it looks fine. For more input, I will have to really look at it when I am felling well enough.

File: settings.yml
Comment (UncleGoogle) [inline settings.yml:64]: Personally I prefer to keep history clean and avoid merge commits (we require to rebase / rebase&squash before merging).

But I see that https://github.com/FriendsOfGalaxy/galaxy-integrations-synchronizer uses merge commits and I wouldn't be so brave to change it right now.

If I had to choose I would I would "subgroup" for synchronized plugins and keep this field value `true`

File: settings.yml
Comment (UncleGoogle) [inline settings.yml:68]: why not? I is quite useful to being able to rebase using UI.

File: settings.yml
Comment (UncleGoogle) [inline settings.yml:166]: nice option! But if I understand correctly - admin can still disable this option in github UI and them merge manually.

File: settings.yml
Comment (UncleGoogle) [inline settings.yml:150]: remember to not register safe-settings app before we invite more people :)

File: settings.yml
Comment (UncleGoogle) [inline settings.yml:171]: I'm not sure how it works. Does it mean we can only add new commits via PRs?

**Review by UncleGoogle — CHANGES_REQUESTED** [review]
Looks like it is impossible to change admin/owner policies, so yeah - you can always disable app bot or change rules manually and do bad things before the app will synchronize back (if it running on cron). 

What we gain with this app is:
- automation to change rules on different repos
- ability to change admin-only settings by NOT being an admin (push rights)

So the most important thing for me is that I can be a normal push-rights user from github perspective and still be able to change rules if other ppl like me want to change some admin stuff. 

From the security point of view it opens a door for privilege escalation from write permission to admin permissions by approving and merging change to this `admin/settings.yaml`. So push user can add himself to "collaborators" and set an "admin" role. But if everyghing is setup correctly it won't be possible without X approvals, so we should be fine :)

So to sum up, by design it is safe when:
- rights to THIS repository are strict
- merge request approval rules are setup correctly
- the installed app works as intended (yeah - it is still a dependency so any bug there may hurts us!)
- we understand all the options well
- there is as little administrators as possible (shouldn't be a problem as we no longer need them to change rules)

A very good thing we could do with setup is to set organization owner to artificial github account and use it only in the case of any emergency... Then set 0 administrators. Then we all can safely use our private accounts on "push" right here. The bad side is that all stuff can take longer time. But at the and this is good - for example we don't won't to invite other people quickly here. Invitation needs some approvals as it is equal to giving 1/3 of admin rights to someone.

But for now it will be enough to have 2FA setup correctly and follow basic security rules (no public wifi's, make your router safe, don't be phished... as OTP won't  help in case of phishing attacks). Maybe that's all:D

File: settings.yml
Comment (urwrstkn8mare) [reply settings.yml:60]: Oh yeah true I'll go ahead and change it.

File: settings.yml
Comment (urwrstkn8mare) [reply settings.yml:68]: Well I initially wanted to do this so there is a clear history on what's happened to the code and also what PR it came from. I was concerned there were some security implications on changing the repo's history. For example if in some documentation it linked to a commit which no longer exists. That may be confusing/annoying. That's not a security issue and now that I think about it I can't really think of security issues. But I may have the idea of rebasing in my head wrong.

File: settings.yml
Comment (urwrstkn8mare) [reply settings.yml:64]: Oh interesting but I guess having the option isn't really harmful.

File: settings.yml
Comment (urwrstkn8mare) [reply settings.yml:166]: Yeah but at least it incentivises getting the required approvals first.

File: settings.yml
Comment (urwrstkn8mare) [reply settings.yml:171]: Yeah I'm pretty sure. Haven't tried it but I guess we'll see.

**Comment by urwrstkn8mare** [comment]
@UncleGoogle Thanks for the feedback I've responded to basically everything you've had to say. I'm pretty sure we're all able to edit the files to this PR so please feel free to change and add things as you please.

File: settings.yml
Comment (UncleGoogle) [reply settings.yml:166]: yup!

File: settings.yml
Comment (UncleGoogle) [reply settings.yml:68]: > But I may have the idea of rebasing in my head wrong.

Indeed, but github docs are misleading a bit about that (https://github.com/github/docs/issues/17959). It won't change existing history on our repository in the case someone has a PR from his fork - it will change history of the feature/fix branch on HIS fork (in general: head branch), so this definitely not an intrusive way interacting with our base branch.

This is more less equivalent to workflow like that:
```bash
git checkout -b new_stuff
(hack hack & commit)
git fetch origin main
git rebase origin/main
git push -f   # rewrite history on new_stuff branch before we can cleanly add all commits from new_stuff on the top of main
git checkout main
git merge new_stuff --ff-only
```

File: settings.yml
Comment (UncleGoogle) [reply settings.yml:64]: It isn't... most of the time. In more complicated flows like git-flow it can add some headache when merging back and forth between release branch, master and feature, but that won't be our case I hope...

you're right, I'll add PR when have historically-proved arguments, not only opinions:)

**Comment by UncleGoogle** [comment]
> I'm pretty sure we're all able to edit the files to this PR

sure, but I don't used to use PRs like that... usually the author is responsible for making changes and adhering or opposing the comments, otherwise each reviewer could change stuff he don't like and who will then review that changes? :confused: 

Sorry if you feel I'm nitpicking (I don't know, just impression). It is not crucial for now, so if the defaults are to allow all types of merge allowed, my proposal is to not disable any of them now.

File: settings.yml
Comment (AndrewDWhite) [reply settings.yml:68]: This article may help you to decide what you prefer. I have used the merge approach more often, but either can work. The larger the group of developers working in parallel the more often I have used the merge approach.
https://www.atlassian.com/git/articles/git-team-workflows-merge-or-rebase

Comment (urwrstkn8mare) [reply settings.yml:68]: Yeah I think this can be discussed more later but for the purpose of the safe-settings config, I think allowing all 3 seems about right.

**Comment by urwrstkn8mare** [comment]
> The bad side is that all stuff can take longer time. But at the and this is good - for example we don't won't to invite other people quickly here. Invitation needs some approvals as it is equal to giving 1/3 of admin rights to someone.

Actually I'm pretty sure we can setup this repo to have more strict settings on approval requirements. So every PR must be approved by two members from an admin team or something of that sort. 

> sure, but I don't used to use PRs like that... usually the author is responsible for making changes and adhering or opposing the comments, otherwise each reviewer could change stuff he don't like and who will then review that changes? 😕

Yeah actually this is a bit of a weird format I realised, I'll just merge it and if you find anything out or have any suggestions feel free to create another PR or discussion. 

> Sorry if you feel I'm nitpicking (I don't know, just impression).

Not at all appreciate the detailed feedback and opinions you have to offer. :)

> It is not crucial for now, so if the defaults are to allow all types of merge allowed, my proposal is to not disable any of them now.

100% agree

**Pull request merged**
